aten/src/THCUNN/SpatialUpSamplingBicubic.cu

```diff
@@ -0,0 +1,243 @@
+#include "THCUNN.h"
+#include "THCTensor.hpp"
+#include "common.h"
+#include "linear_upsampling.h"
+#include "THCDeviceTensor.cuh"
+#include "THCDeviceTensorUtils.cuh"
+#include "THCDeviceUtils.cuh"
+#include "THCHalf.h"
+#include "THCHalfAutoNumerics.cuh"
+#include "THCAtomics.cuh"
+
+__constant__ int size = (1 << 10);
```

Comment:
I'd prefer move all these helper functions to a separate file so that it can be shared between cubic, bicubic etc. You could rename linear_upsampling.h to include these.